[5.6] Fix ordering of migrations in status command

tomgrohl opened the pull request:
First pull request so bear with me.

This relates to issue #24107 that I opened.

This PR uses the ran migrations and diffs with all the migration files to get any unran ones.

The resulting array thats passed to the table is ordered correctly by name and batch

tomgrohl committed:
Fix ordering of migrations in status command

GrahamCampbell commented on src/Illuminate/Database/Console/Migrations/StatusCommand.php:
```diff
@@ -75,14 +75,23 @@ public function handle()
+                return $this->migrator->getMigrationName($migration);
+            })->diff($ran);
+
+        foreach ($diff as $migrationName) {
```
Why the foreach, and not just another map?

tomgrohl replied:
True. I'll replace the foreach's with collections

GrahamCampbell commented:
Thanks for the PR. I've left a review comment. :)

tomgrohl committed:
Use collections over foreach

tomgrohl committed:
Stye CI fixes

tomgrohl commented:
I've updated it to use collections instead of foreach's. Not sure if thats any better

GrahamCampbell commented:
Could you add a test that failed before, but passes after please. Then I'd be more confident this works. :)

tomgrohl commented:
Sure leave it with me.

tomgrohl committed:
Fix parameter order. Add unit test for status command

tomgrohl committed:
Style CI fixes

tomgrohl commented:
Added a test for the change, test did fail originally because I had the parameters the wrong way round on the first map call. This is now fixed.

I've used Symfony's CommandTester so I could add an assertion for the table output from the command. I ran the test with the original`getStatusFor` method, which fails and then afterwards with my change, which passes.

GrahamCampbell commented:
Great, this looks solid. 👍

taylorotwell commented:
Broken on my machine:

![image](https://user-images.githubusercontent.com/463230/39706930-82d5b5b0-51d8-11e8-93ed-b7605e2d0601.png)

tomgrohl commented:
Can I assume because his has been closed you don't want me to investigate and fix?

tomgrohl commented:
@X-Coder264 Ah thanks. I'll do that.

tomgrohl commented:
@taylorotwell any chance you can run the command with `-v` so I can try and debug the issue please? I'm having trouble replicating it.

tomgrohl commented:
I'm having trouble replicating the error @taylorotwell got. I've tried it with different migration states, unran + no ran, unran, ran, no migrations with multiple batches.

Usually and `Array to string conversion` error with an `array_diff` happens if an array is multi-dimensional, but as far as I can see there not.
